src/ILVerify/tests/ILMethodTester.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Internal.IL;
+using Internal.TypeSystem.Ecma;
+using Xunit;
+
+namespace ILVerify.Tests
+{
+    abstract class TestCase
+    {
+        public string MethodName { get; set; }
+        public ILImporter ILImporter { get; set; }
+    }
+
+    /// <summary>
+    /// Describes a test case with a method that contains valid IL
+    /// </summary>
+    class ValidILTestCase : TestCase { }
+
+    /// <summary>
+    /// Describes a test case with a method that contains invalid IL with the expected VerifierErrors
+    /// </summary>
+    class InvalidILTestCase : TestCase
+    {
+        public List<VerifierError> ExpectedVerifierError { get; set; }
+
+        /// <summary>
+        /// The expected exception during validation.
+        /// If no exception is thrown during verification this must be null.
+        /// </summary>
+        public bool ExpectsException { get; set; }
+    }
+
+    /// <summary>
+    /// Parses the methods in the test assemblies. 
+    /// It loads all assemblies from the test folder defined in <code>TestDataLoader.TESTASSEMBLYPATH</code>
+    /// This class feeds the xunit Theories
+    /// </summary>
+    class ILReader
```

Comment:
`ILReader` does not match what this type does - it does not read IL.

Would it make sense to merge it with TestDataLoader ?